contrib/babelfishpg_tsql/src/pltsql_ruleutils.c

```diff
@@ -0,0 +1,2302 @@
+ * as a parameter, and append their text output to its contents.
+ * ----------
+ */
+
+static char *deparse_expression_pretty(Node *expr, List *dpcontext,
+									   bool forceprefix, bool showimplicit,
+									   int prettyFlags, int startIndent);
+static char *generate_qualified_relation_name(Oid relid);
+static char *generate_qualified_type_name(Oid typid);
+static void get_rule_expr(Node *node, deparse_context *context,
+						  bool showimplicit);
+static char *get_variable(Var *var, int levelsup, bool istoplevel,
+						  deparse_context *context);
+static void get_special_variable(Node *node, deparse_context *context,
+								 void *callback_arg);
+static void resolve_special_varno(Node *node, deparse_context *context,
+								  rsv_callback callback, void *callback_arg);
+static void simple_quote_literal(StringInfo buf, const char *val);
+static void get_const_expr(Const *constval, deparse_context *context,
+						   int showtype);
+static void get_const_collation(Const *constval, deparse_context *context);
+static void get_coercion_expr(Node *arg, deparse_context *context,
+							  Oid resulttype, int32 resulttypmod,
+							  Node *parentNode);
+static void set_deparse_plan(deparse_namespace *dpns, Plan *plan);
+static void get_oper_expr(OpExpr *expr, deparse_context *context);
+static void get_func_expr(FuncExpr *expr, deparse_context *context,
+						  bool showimplicit);
+static char *get_relation_name(Oid relid);
+static char *generate_operator_name(Oid operid, Oid arg1, Oid arg2);
+static char *generate_function_name(Oid funcid, int nargs,
+									List *argnames, Oid *argtypes,
+									bool has_variadic, bool *use_variadic_p,
+									ParseExprKind special_exprkind);
+static void push_child_plan(deparse_namespace *dpns, Plan *plan,
+							deparse_namespace *save_dpns);
+static void pop_child_plan(deparse_namespace *dpns,
+						   deparse_namespace *save_dpns);
+static void add_cast_to(StringInfo buf, Oid typid);
+static const char *get_simple_binary_op_name(OpExpr *expr);
+static bool isSimpleNode(Node *node, Node *parentNode, int prettyFlags);
+static Plan *find_recursive_union(deparse_namespace *dpns,
+								  WorkTableScan *wtscan);
+static text *string_to_text(char *str);
+static char *tsql_get_constraintdef_worker(Oid constraintId, bool fullCommand,
+										 int prettyFlags, bool missing_ok);
+
+/*
+ * tsql_get_constraintdef
+ *
+ * Returns the definition for the constraint, ie, everything that needs to
+ * appear after "ALTER TABLE ... ADD CONSTRAINT <constraintname>".
+ */
+Datum
+tsql_get_constraintdef(PG_FUNCTION_ARGS)
+{
+	Oid			constraintId = PG_GETARG_OID(0);
+	int			prettyFlags;
+	char	   *res;
+
+	prettyFlags = PRETTYFLAG_INDENT;
+
+	res = tsql_get_constraintdef_worker(constraintId, false, prettyFlags, true);
+
+	if (res == NULL)
+		PG_RETURN_NULL();
+
+	PG_RETURN_TEXT_P(string_to_text(res));
+}
+
+PG_FUNCTION_INFO_V1(tsql_get_constraintdef);
+
+Datum
+tsql_get_constraintdef_ext(PG_FUNCTION_ARGS)
```

Comment:
This isnt being called anywhere, whats the purpose for adding this?

Reply:
This can be removed in the next commit. We don't need this function.